src/optimize.jl

```diff
@@ -68,26 +68,16 @@ function optimize(
     opt_st = maybe_init_optimizer(state_init, optimizer, params)
     obj_st = maybe_init_objective(state_init, rng, objective, problem, params, restructure)
     avg_st = maybe_init_averager(state_init, averager, params)
-    grad_buf = DiffResults.DiffResult(zero(eltype(params)), similar(params))
     stats = NamedTuple[]
 
     for t in 1:max_iter
         stat = (iteration=t,)
 
-        grad_buf, obj_st, stat′ = estimate_gradient!(
-            rng,
-            objective,
-            adtype,
-            grad_buf,
-            problem,
-            params,
-            restructure,
-            obj_st,
-            objargs...,
+        grad, obj_st, stat′ = estimate_gradient(
```

Comment:
keeping the inplace variant here would likely be good for perf long term [and avoid re allocating new arrays for the gradient storage every iteration, if supported]

Reply:
My reasoning was that to compute that gradient, we allocate a ton inside anyways, so I thought this wouldn't make a noticeable difference. What's your experience?

Reply:
I'm not sure its the bottleneck atm, but if enough other stuff gets optimized it might become one down the line. Plus if it's already there with the fast version, might as well keep it rather than intentionally slowing it down

Reply:
Yeah fair enough.